crates/lakekeeper/src/service/storage/s3.rs

```diff
@@ -747,6 +752,26 @@ impl S3Profile {
             assume_role_builder
         };
 
+        let assume_role_builder = if let Some(session_tags) = &self.sts_session_tags {
+            if !session_tags.is_empty() {
+                let tags: Vec<Tag> = session_tags
+                    .iter()
+                    .map(|(key, value)| {
+                        Tag::builder()
```

Comment:
Tag actually has a infallible constructor: `Tag { key, value}`. The builder is fallible.
Using it would avoid error handling / panic